deploy/platform/kubernetes/feature-agent/resources.yaml

```diff
@@ -88,14 +97,23 @@ spec:
   template:
     metadata:
       labels:
+        # @feature: java-agent-injector; enable the java agent injector
+        swck-java-agent-injected: "true"
         app: songs
       annotations:
         sidecar.istio.io/inject: "${AGENTLESS}"
+        # @feature: java-agent-injector; set the java agent configuration
+        strategy.skywalking.apache.org/inject.Container: "songs"
+        strategy.skywalking.apache.org/agent.Overlay: "true"
+        agent.skywalking.apache.org/agent.service_name: "agent::songs"
     spec:
       containers:
         - name: songs
           image: ${HUB}/songs-service:${TAG}
           imagePullPolicy: Always
+          command: ["java"]
+          # @feature: java-agent-injector; we need to add $(AGENT_OPTS) to enable the injected configuration 
+          args: ${JAVA_AGENT}
```

Comment:
Why do we need this? Does `swck` requires users to modify these?

Reply:
Yep, the injector needs to add the `AGENT_OPTS` to JVM argument like `["$(AGENT_OPTS)","-jar","/app.jar"]`.

Reply:
> Yep, the injector needs to add the `AGENT_OPTS` to JVM argument like `["$(AGENT_OPTS)","-jar","/app.jar"]`.

Sorry I didn't follow the related pull request in swck repo, but why don't we use environment name `JAVA_TOOL_OPTIONS` so that JVM can automatically pick this up and users don't need to modify their command?

Reply:
Make sense to me, we can improve it in the next version.

Reply:
> Make sense to me, we can improve it in the next version.

Nice 🎉

Reply:
Now the injector is using the [way](https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/setting-override/#agent-options) to inject. I don't know much about java, can it be used as an environment variable? if possible, then we don’t need to add it to the startup parameters.

Reply:
> Now the injector is using the [way](https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/setting-override/#agent-options) to inject. I don't know much about java, can it be used as an environment variable? if possible, then we don’t need to add it to the startup parameters.

@dashanji yes, take a look at this https://github.com/apache/skywalking-java/blob/96ea77cf8f1fa922ed4f1e8c4bf9796e33790bdf/Dockerfile#L31 , this env var is automatically picked up by JVM and it's the same effect as `java $JAVA_TOOL_OPTIONS /app.jar` for all Java applications in that container

Reply:
I learn from it, thanks a lot.